docs/server/source/http-client-server-api.rst

```diff
@@ -671,7 +674,7 @@ Blocks
 
    :resheader Content-Type: ``application/json``
 
-   :statuscode 200: A list of blocks containing a transaction with ID ``transaction_id`` was found and returned.
+   :statuscode 200: A list containing at most one block height where ``transaction_id`` was found and returned.
```

Comment:
I see that you added the CRITICAL log message if the response list has more than one block ID. Good!

If there's more than one block with the given transaction_id, I guess what will happen is that the HTTP response will contain a list with more than one block ID, and the response status code will be 200. Maybe the description of that 200 status code could be changed a bit, to reflect that possibility.

I think the status code 200 can also happen if there are no blocks containing a transaction with that transaction ID.

Maybe:

```text
   :statuscode 200: The request was properly formed and zero or more blocks were found containing the specified ``transaction_id``.
```

Also, maybe the sentence describing this endpoint ("Retrieve a list of block IDS...") could be followed by a second sentence noting that the list should be either empty or contain only one block ID.